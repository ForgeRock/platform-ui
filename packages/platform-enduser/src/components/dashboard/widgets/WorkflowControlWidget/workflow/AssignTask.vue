<!-- Copyright (c) 2020-2021 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <BCardBody class="pt-3">
    <BRow>
      <BCol
        lg="8"
        offset-lg="1">
        <div class="form-group row">
          <label
            for="assignSelect"
            class="col-sm-2 col-form-label">
            {{ $t('pages.workflow.assignTo') }}
          </label>
          <div class="col-sm-10">
            <div class="d-flex">
              <BFormSelect
                v-model="selected"
                name="assignSelect"
                :options="assigneeOptions"
                class="mb-3 mr-2" />
              <BButton
                variant="primary"
                class="mb-3 d-flex align-self-end"
                @click="assignTask">
                {{ $t('pages.workflow.assign') }}
              </BButton>
            </div>
          </div>
        </div>
        <div class="form-group row">
          <div class="col-sm-2 col-form-label">
            {{ $t('pages.workflow.details') }}
          </div>
          <div class="col-sm-10">
            <BCard>
              <dl class="row m-0">
                <template v-for="(detail, index) in taskDetailsList">
                  <dt
                    :key="`taskname-${index}-${uniqueId}`"
                    class="col-6">
                    {{ detail.name }}
                  </dt>
                  <dd
                    :key="`taskvalue-${index}-${uniqueId}`"
                    class="col-6">
                    {{ detail.value || $t('pages.workflow.notAvailable') }}
                  </dd>
                </template>
              </dl>
            </BCard>
          </div>
        </div>
      </BCol>
    </BRow>
  </BCardBody>
</template>

<script>
/* eslint-disable no-underscore-dangle */
import { isEmpty } from 'lodash';
import axios from 'axios';

/**
* @description Dashboard widget that allows a user to assign a task
*
* */
export default {
  name: 'AssignTask',
  props: {
    taskDefinition: {
      type: Object,
      default: () => {},
    },
    shown: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      taskDetailsList: [],
      selected: this.$store.state.UserStore.userName,
      uniqueId: null,
    };
  },
  mounted() {
    this.uniqueId = this._uid;
  },
  computed: {
    assigneeOptions() {
      const loggedUserName = this.$store.state.UserStore.userName;

      if (!isEmpty(this.taskDefinition.task.usersToAssign)) {
        return this.taskDefinition.task.usersToAssign.map(({ username, displayableName }) => {
          const value = username;
          const text = username === loggedUserName ? this.$t('pages.workflow.me') : displayableName;
          return { value, text };
        });
      }
      return [{ value: loggedUserName, text: this.$t('pages.workflow.me') }];
    },
  },
  methods: {
    assignTask() {
      this.$emit('assignTask', { id: this.taskDefinition.task._id, assignee: this.$store.state.UserStore.userId });
    },
    generateDisplayDetails(formProperties, variables) {
      return formProperties.reduce((acc, property) => acc.concat({ _id: property._id, name: property.name, value: variables[property._id] }), []);
    },
  },
  watch: {
    shown(val) {
      if (val
                && (this.taskDefinition.process.processDefinition === null || this.taskDefinition.process.processDefinition === undefined)
                && this.taskDetailsList.length === 0) {
        const workflowInstance = axios.create({
          baseURL: process.env.VUE_APP_IDM_URL,
          timeout: 5000,
          headers: {},
        });
        workflowInstance.get(`/workflow/processdefinition/${this.taskDefinition.task.processDefinitionId}`).then((processDetails) => {
          this.taskDetailsList = this.generateDisplayDetails(processDetails.data.formProperties, this.taskDefinition.task.variables);
        });
      } else if (this.taskDetailsList.length === 0) {
        this.taskDetailsList = this.generateDisplayDetails(this.taskDefinition.process.processDefinition.formProperties, this.taskDefinition.task.variables);
      }
    },
  },
};
</script>
