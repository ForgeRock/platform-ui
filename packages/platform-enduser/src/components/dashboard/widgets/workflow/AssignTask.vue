<template>
  <BCardBody class="pt-3">
    <BRow>
      <BCol
        lg="8"
        offset-lg="1"
      >
        <div class="form-group row">
          <label class="col-sm-2 col-form-label">
            {{ $t('pages.workflow.assignTo') }}
          </label>
          <div class="col-sm-10">
            <div class="d-flex">
              <BFormSelect
                v-model="selected"
                :options="assigneeOptions"
                class="mb-3 mr-2"
              />
              <BButton
                type="button"
                variant="primary"
                class="mb-3 d-flex align-self-end"
                @click="assignTask"
              >
                {{ $t('pages.workflow.assign') }}
              </BButton>
            </div>
          </div>
        </div>
        <div class="form-group row">
          <label class="col-sm-2 col-form-label">
            Details
          </label>
          <div class="col-sm-10">
            <BCard>
              <dl class="row m-0">
                <template v-for="(detail, index) in taskDetails">
                  <dt
                    :key="`taskname-${index}`"
                    class="col-6"
                  >
                    {{ detail.name }}
                  </dt>
                  <dd
                    :key="`taskvalue-${index}`"
                    class="col-6"
                  >
                    {{ detail.value }}
                  </dd>
                </template>
              </dl>
            </BCard>
          </div>
        </div>
      </BCol>
    </BRow>
  </BCardBody>
</template>

<script>
import _ from 'lodash';

/**
* @description Dashboard widget that allows a user to assign a task
*
* */
export default {
  name: 'AssignTask',
  props: {
    taskDefinition: {
      type: Object,
      default: () => {},
    },
  },
  data() {
    return {
      workflowService: null,
      selected: this.$root.userStore.state.userName,
    };
  },
  computed: {
    process() {
      return this.taskDefinition.process;
    },
    processDefinition() {
      if (this.process.processDefinition === null) {
        this.$emit('loadProcess', this.process);
      }
      return this.process.processDefinition;
    },
    formProperties() {
      return !_.isNull(this.processDefinition) ? this.processDefinition.formProperties : [];
    },
    task() {
      return this.taskDefinition.task;
    },
    id() {
      /* eslint no-underscore-dangle: ["error", { "allow": ["_id"] }] */
      return this.task._id;
    },
    taskDetails() {
      return this.formProperties.reduce((acc, property) => acc.concat({ _id: property._id, name: property.name, value: this.task.variables[property._id] }), []);
    },
    assigneeOptions() {
      const loggedUserName = this.$root.userStore.state.userName;

      if (!_.isEmpty(this.task.usersToAssign)) {
        return this.task.usersToAssign.map(({ username, displayableName }) => {
          const value = username;
          const text = username === loggedUserName ? this.$t('pages.workflow.me') : displayableName;

          return { value, text };
        });
      }
      return [{ value: loggedUserName, text: this.$t('pages.workflow.me') }];
    },
  },
  methods: {
    assignTask() {
      this.$emit('assignTask', { id: this.id, assignee: this.selected });
    },
  },
};
</script>
