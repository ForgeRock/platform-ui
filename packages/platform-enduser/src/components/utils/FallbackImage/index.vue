<!-- Copyright (c) 2020-2021 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div>
    <BImg
      v-if="imageFound"
      :src="src"
      :width="width"
      :height="width"
      :alt="alt" />
    <span
      v-else
      class="icon-holder p-2 d-flex bg-light border rounded">
      <FrIcon
        :class="['md-24 text-dark mt-auto', inputClass]"
        :name="fallback"
      />
    </span>
  </div>
</template>

<script>
import axios from 'axios';
import FrIcon from '@forgerock/platform-shared/src/components/Icon';

/**
 * @description Component that will load an image and if the image fails to load it will display a fallback image rather than loading nothing
 *
 * */
export default {
  name: 'ImageFallback',
  components: {
    FrIcon,
  },
  props: {
    inputClass: {
      type: String,
      default: () => '',
    },
    src: {
      type: String,
      default: () => '',
    },
    width: {
      type: [String, Number],
      default: () => '',
    },
    alt: {
      type: String,
      default: () => '',
    },
    fallback: {
      type: String,
      default: () => '',
    },
  },
  data() {
    return {
      imageFound: false,
    };
  },
  mounted() {
    if (this.src) {
      axios.get(`${this.src}`).then(({ status }) => {
        this.imageFound = status === 200;
      }).catch((error) => {
        if (error) {
          this.imageFound = false;
        }
      });
    } else {
      this.imageFound = false;
    }
  },
};
</script>

<style language="scss" scoped>
.icon-holder {
  min-height: 2.28rem;
  min-width: 2.28rem;
}
</style>
