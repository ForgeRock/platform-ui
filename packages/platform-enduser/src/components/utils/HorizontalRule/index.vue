<!-- Copyright 2019 ForgeRock AS. All Rights Reserved

Use of this code requires a commercial software license with ForgeRock AS.
or with one of its affiliates. All use shall be exclusively subject
to such license between the licensee and ForgeRock AS. -->
<template>
  <div class="row">
    <div class="col pr-0">
      <hr>
    </div>
    <div
      class="col-auto pt-1"
      v-html="insert">
      {{ insert }}
    </div>
    <div class="col pl-0">
      <hr>
    </div>
  </div>
</template>

<script>
/**
 * @description Component for making a horizontal rule line with text in the middle
 *
 * */
export default {
  name: 'HorizontalRule',
  props: {
    insert: {
      type: String,
      default: () => '',
    },
  },
};
</script>

<style lang="scss" scoped>
</style>
