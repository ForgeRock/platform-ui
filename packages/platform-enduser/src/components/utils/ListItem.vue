<template>
  <div
    v-if="collapsible"
    class="collapsible">
    <BListGroupItem
      href="#"
      :class="[{'list-item-cursor': collapsible===false}]"
      v-b-toggle="toggleId">
      <div class="media">
        <slot name="list-item-header" />
      </div>
    </BListGroupItem>

    <BCollapse
      :id="id"
      :visible="panelShown"
      @hide="$emit('hide')"
      @show="$emit('show')"
      @hidden="$emit('hidden')"
      @shown="$emit('shown')">
      <BCardBody class="pt-3">
        <slot name="list-item-collapse-body" />
      </BCardBody>
    </BCollapse>
  </div>

  <div
    v-else
    @click="$emit('row-click')"
    :class="[{'fr-hover-item': hoverItem}]">
    <BListGroupItem class="noncollapse">
      <div class="media">
        <slot name="list-item-header" />
      </div>
    </BListGroupItem>

    <BCardBody
      v-if="panelShown"
      class="pt-3">
      <slot name="list-item-collapse-body" />
    </BCardBody>
  </div>
</template>

<script>
/**
 * @description Used in conjunction with ListGroup.vue, this is the individual item in each list display.
 *
 * */
export default {
	name: 'ListItem',
	props: {
		collapsible: {
			type: Boolean,
			default: false,
		},
		panelShown: {
			type: Boolean,
			default: false,
		},
		hoverItem: {
			type: Boolean,
			default: false,
		},
	},
	data() {
		return {
			id: null,
		};
	},
	beforeMount() {
		// eslint-disable-next-line no-underscore-dangle
		this.id = `listItem${this._uid}`;
	},
	computed: {
		toggleId() {
			if (this.collapsible) {
				return this.id;
			}
			return null;
		},
	},
};
</script>
<style lang="scss" scoped>
    .media {
        height: 38px;

        & > * {
            -ms-flex-item-align: center;
            align-self: center;
        }
    }

    .list-item-cursor {
        cursor: default;
    }

    .fr-hover-item:hover {
        cursor: pointer;

        .list-group-item {
            background-color: $fr-hover-list-color;
        }
    }

    .collapsible:last-of-type > .list-group-item.collapsed {
        border-bottom-right-radius: $border-radius;
        border-bottom-left-radius: $border-radius;
        border-bottom: 1px solid $border-color;
    }

    .list-group-item-action {
        color: inherit;

        .caret {
            .caret-up {
                display: none;
            }
        }
        .meta {
            line-height: 1.2;
        }
        .btn-cancel {
            display: none;
        }
        .btn-edit {
            display: block;
        }

        &:not(.collapsed) {
            button {
                display: block;
            }
            .btn-cancel {
                display: block;
            }
            .btn-edit {
                display: none;
            }
            .caret {
                .caret-up {
                    display: block;
                }
                .caret-down {
                    display: none;
                }
            }
        }
    }

    .list-group-item-action:not(.collapsed),
    .list-group-item-action:not(.collapsed):hover,
    .list-group-item-action:not(.collapsed):focus,
    .noncollapse {
        background-color: $card-bg;
        border-bottom-color: transparent;
        cursor: initial;
    }

</style>
