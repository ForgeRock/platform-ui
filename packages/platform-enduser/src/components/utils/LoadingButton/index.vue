<!--
Copyright (c) 2020 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details.
-->

<template>
  <button
    :class="['btn btn-primary d-flex align-items-center', {'disabled': loading}]"
    type="button"
    @click="$emit('click')">
    <span :class="[{'align-self-start': !large, 'm-auto': large}]">
      {{ label }}
    </span>
    <span :class="['fr-grow', {'fr-grow-in': loading}]">
      <Transition name="fr-fade">
        <ClipLoader
          v-if="loading"
          :color="'white'"
          :size="'1rem'"
          class="position-relative fr-clip-loader ml-3" />
      </Transition>
    </span>
  </button>
</template>
<script>
import { ClipLoader } from 'vue-spinner/dist/vue-spinner.min';
/**
 * @description Button with built in loading spinner that can be used to show users something is loading
 *
 * */
export default {
  name: 'LoadingButton',
  components: { ClipLoader },
  props: {
    label: {
      type: String,
      default: () => '',
    },
    loading: {
      type: Boolean,
      default: false,
    },
    large: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
<style lang="scss" scoped>
.fr-grow {
  width: 0;
  transition: all 0.25s ease;
}

.fr-grow-in {
  width: 3rem;
}

.fr-fade-enter-active {
  transition: all 0.125s ease;
}

.fr-fade-leave {
  transition: all 0.125s ease;
}

.fr-fade-enter,
.fr-fade-leave-to {
  opacity: 0;
}

.fr-fade-enter-to {
  opacity: 1;
}

.fr-clip-loader {
  top: 0.15rem;
}
</style>
