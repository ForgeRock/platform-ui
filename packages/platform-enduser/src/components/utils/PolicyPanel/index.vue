<template>
  <div class="card border-0 mt-3">
    <Transition
      name="fade"
      mode="out-in"
      :duration="150">
      <template v-if="!isLoading">
        <ul
          class="text-left pl-3"
          v-if="!isValid">
          <li
            v-for="policy in policies"
            :key="policy.policyId"
            :class="[{'fr-valid': !includes(policyFailures, policy.name)}, 'text-primary fr-policy-list-item']">
            <small class="text-body">
              {{ translate(policy) }}
            </small>
          </li>
        </ul>
        <div
          v-else
          class="alert alert-success mt-1"
          role="alert">
          <i class="material-icons-outlined text-success">
            check_circle
          </i>
          {{ $t('common.policyValidationMessages.successMessages.password') }}
        </div>
      </template>
    </Transition>
  </div>
</template>

<script>
import {
  includes,
  isArray,
  isEmpty,
} from 'lodash';

/**
 * @description Part of the password policy component to display the list of policy items required
 *
 * */
export default {
  name: 'PolicyPanel',
  props: {
    policies: {
      type: Array,
      default: () => [],
    },
    policyFailures: {
      type: [String, Array],
      default: () => '',
    },
  },
  data() {
    return {};
  },
  computed: {
    isValid() {
      return isArray(this.policyFailures) && isEmpty(this.policyFailures);
    },
    isLoading() {
      return this.policyFailures === 'loading' || this.policyFailures === false;
    },
  },
  methods: {
    translate(policy) {
      const path = `common.policyValidationMessages.${policy.name}`;

      return this.$t(path, policy.params);
    },
    includes,
  },
  watch: {
    policyFailures(value) {
      this.policyFailures = value;
    },
  },
};
</script>
<style lang="scss" scoped>
.fr-policy-list-item {
  line-height: 1.25;
}

.fr-valid {
  opacity: 0.5;
}
</style>
