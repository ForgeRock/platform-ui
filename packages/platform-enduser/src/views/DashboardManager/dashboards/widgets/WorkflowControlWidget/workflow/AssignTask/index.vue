<!-- Copyright (c) 2020-2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <BCardBody class="pt-3">
    <BRow>
      <BCol
        lg="8"
        offset-lg="1">
        <div class="form-group row">
          <label
            for="assignSelect"
            class="col-sm-2 col-form-label">
            {{ $t('pages.workflow.assignTo') }}
          </label>
          <div class="col-sm-10">
            <div class="d-flex">
              <BFormSelect
                v-model="selected"
                name="assignSelect"
                :options="assigneeOptions"
                class="mb-3 mr-2" />
              <BButton
                variant="primary"
                class="mb-3 d-flex align-self-end"
                @click="assignTask">
                {{ $t('pages.workflow.assign') }}
              </BButton>
            </div>
          </div>
        </div>
        <div class="form-group row">
          <div class="col-sm-2 col-form-label">
            {{ $t('pages.workflow.details') }}
          </div>
          <div class="col-sm-10">
            <BCard>
              <dl class="row m-0">
                <template
                  v-for="(detail, index) in taskDetailsList"
                  :key="`taskname-${index}-${uniqueId}`">
                  <dt class="col-6">
                    {{ detail.name }}
                  </dt>
                  <dd class="col-6">
                    {{ detail.value || $t('pages.workflow.notAvailable') }}
                  </dd>
                </template>
              </dl>
            </BCard>
          </div>
        </div>
      </BCol>
    </BRow>
  </BCardBody>
</template>

<script>
import {
  BButton,
  BCard,
  BCardBody,
  BCol,
  BFormSelect,
  BRow,
} from 'bootstrap-vue';
import { mapState } from 'pinia';
import { useUserStore } from '@forgerock/platform-shared/src/stores/user';
import { isEmpty } from 'lodash';
import getFQDN from '@forgerock/platform-shared/src/utils/getFQDN';
import axios from 'axios';

/**
* @description Dashboard widget that allows a user to assign a task
*
* */
export default {
  name: 'AssignTask',
  components: {
    BButton,
    BCard,
    BCardBody,
    BCol,
    BFormSelect,
    BRow,
  },
  props: {
    taskDefinition: {
      type: Object,
      default: () => {},
    },
    shown: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    const userStore = useUserStore();

    return {
      taskDetailsList: [],
      selected: userStore.userName,
      uniqueId: null,
    };
  },
  mounted() {
    this.uniqueId = this._uid;
  },
  computed: {
    ...mapState(useUserStore, ['userId', 'userName']),
    assigneeOptions() {
      const loggedUserName = this.userName;

      if (!isEmpty(this.taskDefinition.task.usersToAssign)) {
        return this.taskDefinition.task.usersToAssign.map(({ username, displayableName }) => {
          const value = username;
          const text = username === loggedUserName ? this.$t('pages.workflow.me') : displayableName;
          return { value, text };
        });
      }
      return [{ value: loggedUserName, text: this.$t('pages.workflow.me') }];
    },
  },
  methods: {
    assignTask() {
      this.$emit('assignTask', { id: this.taskDefinition.task._id, assignee: this.userId });
    },
    generateDisplayDetails(formProperties, variables) {
      return formProperties.reduce((acc, property) => acc.concat({ _id: property._id, name: property.name, value: variables[property._id] }), []);
    },
  },
  watch: {
    shown(val) {
      if (val
                && (this.taskDefinition.process.processDefinition === null || this.taskDefinition.process.processDefinition === undefined)
                && this.taskDetailsList.length === 0) {
        const workflowInstance = axios.create({
          baseURL: getFQDN(process.env.VUE_APP_IDM_URL),
          headers: {},
        });
        workflowInstance.get(`/workflow/processdefinition/${this.taskDefinition.task.processDefinitionId}`).then((processDetails) => {
          this.taskDetailsList = this.generateDisplayDetails(processDetails.data.formProperties, this.taskDefinition.task.variables);
        });
      } else if (this.taskDetailsList.length === 0) {
        this.taskDetailsList = this.generateDisplayDetails(this.taskDefinition.process.processDefinition.formProperties, this.taskDefinition.task.variables);
      }
    },
  },
};
</script>
