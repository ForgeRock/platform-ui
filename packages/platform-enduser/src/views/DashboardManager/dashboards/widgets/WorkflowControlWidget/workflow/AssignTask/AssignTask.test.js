/**
 * Copyright (c) 2020-2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { shallowMount } from '@vue/test-utils';
import { setupTestPinia } from '@forgerock/platform-shared/src/utils/testPiniaHelpers';
import AssignTask from './index';

describe('AssignTask.vue', () => {
  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    setupTestPinia({
      user: {
        userId: 'userId',
        userName: 'myUsername',
      },
    });
    wrapper = shallowMount(AssignTask, {
      props: {
        taskDefinition: {
          task: {
            _id: 123,
            usersToAssign: [
              {
                username: 'John90210',
                displayableName: 'John',
              },
            ],
            variables: {
              123: 'myVariables',
            },
          },
        },
        shown: false,
      },
      global: {
        mocks: {
          $t: (key) => (key),
        },
      },
    });
  });

  it('Emits assignTask object', () => {
    wrapper.vm.assignTask();
    expect(wrapper.emitted().assignTask).toEqual([[{
      id: 123,
      assignee: 'userId',
    }]]);
  });

  it('Generates Display Details', () => {
    const taskVariables = {
      123: 'myVariables',
    };
    const formProperties = [{
      _id: 123,
      name: 'myPropertyName',
    }];

    expect(wrapper.vm.generateDisplayDetails(formProperties, taskVariables)).toEqual([{
      _id: formProperties[0]._id,
      name: formProperties[0].name,
      value: taskVariables['123'],
    }]);
  });

  describe('Generates assignee options', () => {
    it('returns options if taskDefinition.task.usersToAssign prop present', () => {
      expect(wrapper.vm.assigneeOptions).toEqual([{
        value: 'John90210',
        text: 'John',
      }]);
    });

    it('returns options if taskDefinition.task.usersToAssign IS NOT present', async () => {
      await wrapper.setProps({
        taskDefinition: {
          task: {
            usersToAssign: [],
          },
        },
      });
      expect(wrapper.vm.assigneeOptions).toEqual([{
        value: 'myUsername',
        text: 'pages.workflow.me',
      }]);
    });
  });
});
