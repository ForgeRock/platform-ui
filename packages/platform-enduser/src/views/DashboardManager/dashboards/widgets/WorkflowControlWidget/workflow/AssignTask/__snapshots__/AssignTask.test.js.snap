// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignTask.vue Emits assignTask object: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <AssignTask",
    "taskDefinition=",
    {
      "task": {
        "_id": 123,
        "usersToAssign": [
          {
            "displayableName": "John",
            "username": "John90210",
          },
        ],
        "variables": {
          "123": "myVariables",
        },
      },
    },
    "shown=false",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "overlay" with v-bind value \`false\` will render overlay="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BCardBody",
    "class="pt-3"",
    ">",
    "
",
    " at <AssignTask",
    "taskDefinition=",
    {
      "task": {
        "_id": 123,
        "usersToAssign": [
          {
            "displayableName": "John",
            "username": "John90210",
          },
        ],
        "variables": {
          "123": "myVariables",
        },
      },
    },
    "shown=false",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
